src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataMember.cs

```diff
@@ -242,16 +232,13 @@ internal DataContract MemberTypeContract
                 {
                     if (_memberTypeContract == null)
                     {
-                        if (MemberInfo != null)
+                        if (this.IsGetOnlyCollection)
```

Comment:
I'd prefer if we've left the original code and added TODO-NULLABLE to decrease chance of regressing anything

Reply:
Reverted and added a TODO-NULLABLE.